### PR TITLE
feat: Command Line Interface

### PR DESCRIPTION
**Pull request by jdpatt**
# Basic Command Line Interface

Adds a basic CLI for #89 that retains the current behavior of opening pybert with either `python -m pybert` or simply `pybert`.   This pulls out some business logic that was in the view and pulls that back into pybert.  Click has a little bit slower start-up time than the builtin argparse but makes adding/managing sub-commands much easier.  I could see us adding things like view and configuring ami models as a sub command.

**Examples of Usage**

```shell
$ pybert --help
Usage: pybert [OPTIONS] COMMAND [ARGS]...

  Serial communication link bit error rate tester.

Options:
  --version               Show the version and exit.
  -c, --config-file PATH  Load an existing configuration file.
  -r, --results PATH      Load results from a prior run.
  -h, --help              Show this message and exit.

Commands:
  sim  Run a simulation without opening the GUI.
```

- `pybert` - No change from today's behavior
- `pybert -c test_configuration.yaml` - Open the gui but have a configuration already loaded on start-up

**Headless CLI Usage**

```shell
pybert sim --help
Usage: pybert sim [OPTIONS] CONFIG_FILE

  Run a simulation without opening the GUI.

  Will load the CONFIG_FILE from the given filepath, run the simulation and
  then save the results into a file with the same name but a different
  extension as the configuration file.

Options:
  -r, --results PATH  Override the results filename.
  -h, --help          Show this message and exit.
```

- `pybert sim test_configuration.yaml` - Will run the configuration and save it out as test_configuration.pybert_data
- `pybert sim test_configuration.yaml -r out.pybert_data` - Will run the configuration and save it out as out.pybert_data

**Other**

- Caveat that if something is out of the traitsui range, it fails silently. For example, loading a
  configuration with nbits set to 700 instead of the required low of 1000.  Its always been this way just not sure how to improve it without a lot of error checking in PyBertCfg.
- Caveat that user popups on Exceptions for loading/saving are now just messages to not have the GUI
  pop up in CLI mode.
- Removes the `if USE_YAML` to allow loading the pickle config but show it is deprecated.  It does not allow saving
  as pickle for the configuration to force the update to yaml.
- Adds pybert version and timestamp to configuration and results files.
- Script usage is a little easier

```python
from pathlib import Path
from pybert.pybert import PyBERT

config_file = Path("test.yaml")
pybert = PyBERT(run_simulation=True, gui=False)
pybert.load_configuration(config_file)
for bit_rate in range(10,100):
    pybert.bit_rate = bit_rate
    print(f"Simulating bit_rate of {bit_rate} Gbps")
    pybert.simulate()
    pybert.save_results(Path(config_file).with_suffix(".pybert_data").with_name(f"{config_file.stem}_{bit_rate}"))

```

### CODE REVIEW

File: src/pybert/gui/view.py [hunk 1]
```diff
@@ -55,7 +52,7 @@ class MyHandler(Handler):
     def do_run_simulation(self, info):
         """Spawn a simulation thread and run with the current settings."""
         the_pybert = info.object
-        if self.run_sim_thread and self.run_sim_thread.isAlive():
+        if self.run_sim_thread and self.run_sim_thread.is_alive():
```
Comment (capn-freako) [inline src/pybert/gui/view.py:55]: Is this just a name change, or are you pointing to a different function?
(I couldn't find an equivalent change to the `isAlive()` function definition; is that ours or in a library?)

**Review by capn-freako — CHANGES_REQUESTED** [review]
Thanks, David!
Just a few comments/requests this time.
-db

File: src/pybert/gui/view.py (hunk 1 above)
Comment (jdpatt) [reply src/pybert/gui/view.py:55]: Added in 2.6 just to be pep8 or underscore vs. camelCase.

- https://docs.python.org/2.7/library/threading.html#threading.Thread.isAlive

Removed in 3.9

- https://docs.python.org/3/whatsnew/3.9.html

File: pyproject.toml [hunk 2]
```diff
@@ -63,6 +62,9 @@ line-length = 119
 [tool.isort]
 profile = "black"
 
+[tool.docformatter]
+wrap-summaries = 0
```
[the remark this answers is not in the document]
Comment (jdpatt) [reply pyproject.toml:66]: We use longer line lengths than the default, I need to play around with a number to leave it enabled.  I didn't want it randomly changing code outside what I was touching for this PR. (getting better about that!)  I will probably leave it disabled here and attempt to improve it outside of this PR and have one just for formatting.

It was doing weird things to the summary lines in some functions.  Trying to stick with the first line should be a high level summary and then anything after can be multiline.  https://peps.python.org/pep-0257/

File: pyproject.toml [hunk 3]
```diff
@@ -7,14 +7,13 @@ name = "PyBERT"
...
 authors = [ {name = "David Banas",     email = "capn.freako@gmail.com"}
-          , {name = "David Patterson", email = "dpatter877@gmail.com"}
+          , {name = "David Patterson"}
           ]
 urls = { documentation = "https://github.com/capn-freako/PyBERT/wiki"}
 readme = "README.md"
 requires-python = ">=3.8"
 license = {text = "BSD"}
 dependencies = [
-        # "blas>=1.0",  # I thought this might fix my performance test failure.
```
[the remark this answers is not in the document]
Comment (jdpatt) [reply pyproject.toml]: done!

**Review by capn-freako — APPROVED** [review]
Thanks, David!

**Comment by capn-freako** [comment]
Thanks again, David!

So, having _NumPy_ as a `conda_deps` is working for you, at least, for now?

**Pull request merged**